src/Options/ClientOptions.php

```diff
@@ -175,6 +179,7 @@ private function fromArray(array $arr): void
         $this->setLibVersion($arr['libVersion'] ?? null);
         $this->setGapicVersion($arr['gapicVersion'] ?? null);
         $this->setClientCertSource($arr['clientCertSource'] ?? null);
+        $this->setUniverseDomain($arr['universeDomain'] ?? null);
```

Comment:
I do not see value in not initializing UD by null here.
Like if we know that the default value is gonna be `GetUniverseDomainInterface::DEFAULT_UNIVERSE_DOMAIN` lets use it?

Reply:
This is up for debate, but the reason this choice was made is that I wanted the `ClientOptions` to be dumb. If a parameter isn't supplied, it's `null`. Always. They are options, and options are optional, so they're nullable. The client option represent what the user supplied, not what the default values should be. This is important because a user could supply `null`, or they could supply `googleapis.com`, and if we set the default, we wouldn't be able to know if they supplied it or not. Now, that may not matter (and probably doesn't), but that was the rationale here.

Furthermore, I didn't want default values to be defined in `ClientOptions`, but rather by the classes which use them. So in this case, the default universe domain is being defined by `GapicClientTrait`, which I think makes more sense.